src/docker-in-docker/install.sh

```diff
@@ -245,75 +246,77 @@ fi
+        curl -L "https://github.com/docker/compose/releases/download/v${compose_version}/docker-compose-linux-${target_compose_arch}" -o ${docker_compose_path}
+        chmod +x ${docker_compose_path}
+
+        # Download the SHA256 checksum
+        DOCKER_COMPOSE_SHA256="$(curl -sSL "https://github.com/docker/compose/releases/download/v${compose_version}/docker-compose-linux-${target_compose_arch}.sha256" | awk '{print $1}')"
+        echo "${DOCKER_COMPOSE_SHA256}  ${docker_compose_path}" > docker-compose.sha256sum
+        sha256sum -c docker-compose.sha256sum --ignore-missing
+
+        mkdir -p ${cli_plugins_dir}
+        cp ${docker_compose_path} ${cli_plugins_dir}
     fi
+fi
 
-    # Install docker-compose switch if not already installed - https://github.com/docker/compose-switch#manual-installation
-    current_v1_compose_path="$(which docker-compose)"
-    target_v1_compose_path="$(dirname "${current_v1_compose_path}")/docker-compose-v1"
-    if ! type compose-switch > /dev/null 2>&1; then
+# Install docker-compose switch if not already installed - https://github.com/docker/compose-switch#manual-installation
+if [ "${INSTALL_DOCKER_COMPOSE_SWITCH}" = "true" ] && ! type compose-switch > /dev/null 2>&1; then
+    if type docker-compose > /dev/null 2>&1; then
         echo "(*) Installing compose-switch..."
+        current_compose_path="$(which docker-compose)"
+        target_compose_path="$(dirname "${current_compose_path}")/docker-compose-v1"
         compose_switch_version="latest"
         find_version_from_git_tags compose_switch_version "https://github.com/docker/compose-switch"
         curl -fsSL "https://github.com/docker/compose-switch/releases/download/v${compose_switch_version}/docker-compose-linux-${architecture}" -o /usr/local/bin/compose-switch
         chmod +x /usr/local/bin/compose-switch
         # TODO: Verify checksum once available: https://github.com/docker/compose-switch/issues/11
-
         # Setup v1 CLI as alternative in addition to compose-switch (which maps to v2)
-        mv "${current_v1_compose_path}" "${target_v1_compose_path}"
-        update-alternatives --install /usr/local/bin/docker-compose docker-compose /usr/local/bin/compose-switch 99
-        update-alternatives --install /usr/local/bin/docker-compose docker-compose "${target_v1_compose_path}" 1
-    fi
-    if [ "${DOCKER_DASH_COMPOSE_VERSION}" = "v1" ]; then
-        update-alternatives --set docker-compose "${target_v1_compose_path}"
+        mv "${current_compose_path}" "${target_compose_path}"
+        update-alternatives --install ${docker_compose_path} docker-compose /usr/local/bin/compose-switch 99
+        update-alternatives --install ${docker_compose_path} docker-compose "${target_compose_path}" 1
     else
-        update-alternatives --set docker-compose /usr/local/bin/compose-switch
+        err "Skipping installation of compose-switch as docker compose is unavailable..."
```

Comment:
`docker-compose` (v1) is unavailable, but if `docker compose` (v2) is there, wouldn't it make sense to still install compose-switch to get a `docker-compose` executable that converts the command line args and forwards to `docker compose` for compatibility?